xds/internal/xdsclient/xdsresource/filter_chain.go

```diff
@@ -194,10 +203,24 @@ type FilterChainManager struct {
 
 	def *FilterChain // Default filter chain, if specified.
 
+	// Slice of filter chains managed by this filter chain manager.
+	fcs []*FilterChain
+
 	// RouteConfigNames are the route configuration names which need to be
 	// dynamically queried for RDS Configuration for any FilterChains which
 	// specify to load RDS Configuration dynamically.
 	RouteConfigNames map[string]bool
+
+	// Persisted to gracefully close once filter chain manager no longer active.
+	conns []net.Conn
+}
+
+func (fcm *FilterChainManager) AddConn(conn net.Conn) {
+	fcm.conns = append(fcm.conns, conn)
+}
+
+func (fcm *FilterChainManager) Conns() []net.Conn {
+	return fcm.conns
```

Comment:
Maybe this becomes moot if you decide to accept my other comment that the conns should ideally be tracked in the `listenerWrapper` and not in the `FilterChainManager`.

But, if there are indeed valid reasons to track the conns here (that I'm not able to see right now), then what guarantees that accesses to `conns` are serialized? I don't see any locks or any docstrings saying callers should not call `AddConns` and `Conns` concurrently.

Reply:
Wrote an explanation above as to why I will track the Conns in the Filter Chain Manager. Will add documentation that cannot call concurrently.

Reply:
Right now, these calls are protected by listener wrappers mutex.